frontend/components/PhoneinputTest.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { Component, useRef } from "react";
+import {
+  SafeAreaView,
+  StyleSheet,
+  View,
+  StatusBar,
+  TouchableOpacity,
+  Text,
+  Pressable,
+} from 'react-native';
+import PhoneInput from 'react-native-phone-number-input';
+// import {Colors} from 'react-native/Libraries/NewAppScreen';
+// const phoneInput = useRef<PhoneInput>(null);
+type State = {
+    value: string,
+    countryCode: string,
+    formattedValue: string,
+    valid:boolean,
+    disabled: boolean,
+    showMessage: boolean,
+
+}
+class PhoneinputTest extends Component<{}, State> {
+    phoneInput=React.createRef<PhoneInput>();
+    constructor(props: {}) {
+      super(props);
+      this.state = {
+        value: '',
+        countryCode: '',
+        formattedValue: '',
+        valid: false,
+        disabled: false,
+        showMessage: false,
+      }
+    }
+  demo = () =>{
+      console.log("demo triggered")
+  }
+  
+  render(){
+    return (
+        <>
+          <StatusBar barStyle="dark-content" />
+          <View style={styles.container}>
+            <SafeAreaView style={styles.wrapper}>
+              {this.state.showMessage && (
+                <View style={styles.message}>
+                  <Text>Country Code : {this.state.countryCode}</Text>
+                  <Text>Value : {this.state.value}</Text>
+                  <Text>Formatted Value : {this.state.formattedValue}</Text>
+                  <Text>Valid : {this.state.valid ? 'true' : 'false'}</Text>
+                </View>
+              )}
+              <PhoneInput
+                ref={this.phoneInput}
+                defaultValue={this.state.value}
+                defaultCode="IN"
+                layout="first"
+                onChangeText={(text) => { this.setState({value: text}) }}
+                onChangeFormattedText={(text) => {
```

Comment:
whats the difference between text and formatted text? a small explanatory comment or discussion will be useful